tests/periph_spi/main.c

```diff
@@ -0,0 +1,413 @@
+            break;
+        case 2:
+            spiconf.mode = SPI_MODE_2;
+            break;
+        case 3:
+            spiconf.mode = SPI_MODE_3;
+            break;
+        default:
+            CHECK_ASSERT(0, USAGE);
+    }
+
+    if (strcmp(argv[3], "100k") == 0) {
+        spiconf.clk = SPI_CLK_100KHZ;
+    }
+    else if (strcmp(argv[3], "100k") == 0) {
+        spiconf.clk = SPI_CLK_400KHZ;
+    }
+    else if (strcmp(argv[3], "400k") == 0) {
+        spiconf.clk = SPI_CLK_1MHZ;
+    }
+    else if (strcmp(argv[3], "1M") == 0) {
+        spiconf.clk = SPI_CLK_5MHZ;
+    }
+    else if (strcmp(argv[3], "5M") == 0) {
+        spiconf.clk = SPI_CLK_10MHZ;
+    }
+    else {
+        CHECK_ASSERT(0, USAGE);
+    }
+
+    if (port == -1) {                    /* hardware chip select line */
+        spiconf.cs = SPI_HWCS(pin);
+    }
+    else {
+        spiconf.cs = (spi_cs_t)GPIO_PIN(port, pin);
+    }
+    sprintf(printbuf,
+            "spi_acquire(bus=%i, port=%"PRIi32", pin=%"PRIi32", mode=%i, clk=%s)",
+            dev, port, pin, mode, argv[3]);
+    print_cmd(PARSER_DEV_NUM, printbuf);
+    CHECK_ASSERT(spi_acquire(spiconf.dev,
+                             spiconf.cs,
+                             spiconf.mode,
+                             spiconf.clk) == SPI_OK, "ERROR initializing SPI");
+    print_result(PARSER_DEV_NUM, TEST_RESULT_SUCCESS);
+    return 0;
+}
+
+int cmd_spi_release(int argc, char **argv)
+{
+    const char* USAGE = "spi_release DEV";
+    CHECK_ASSERT(argc == 2, USAGE);
+    int dev = sc_arg2dev(argv[1], SPI_NUMOF);
+    CHECK_ASSERT(dev != ARGS_ERROR, USAGE);
+    sprintf(printbuf, "spi_release(dev=%i)", dev);
+    print_cmd(PARSER_DEV_NUM, printbuf);
+    print_result(PARSER_DEV_NUM, TEST_RESULT_SUCCESS);
+    spi_release(dev);
+    return 0;
+}
+
+int cmd_spi_transfer_byte(int argc, char **argv)
+{
+    const char* USAGE = "spi_release DEV CS_PORT CS_PIN CONT OUT";
+    CHECK_ASSERT(argc == 6, USAGE);
+    int dev = sc_arg2dev(argv[1], SPI_NUMOF);
+    int32_t port, pin;
+    int cont;
+    uint8_t out;
+    CHECK_ASSERT(dev != ARGS_ERROR &&
+        (sc_arg2int(argv[4], &cont) == ARGS_OK) &&
+        (sc_arg2s32(argv[2], &port) == ARGS_OK) &&
+        (sc_arg2s32(argv[3], &pin) == ARGS_OK) &&
+        (sc_arg2u8(argv[5], &out) == ARGS_OK), USAGE);
+
+    sprintf(printbuf,
+           "spi_transfer_byte(dev=%i, port=%"PRIi32", pin=%"PRIi32", cont=%i, out=%u)",
+            dev, port, pin, cont, out);
+
+    print_cmd(PARSER_DEV_NUM, printbuf);
+    print_data_int(PARSER_DEV_NUM,
+                   spi_transfer_byte(dev,
+                                     GPIO_PIN(port, pin),
+                                     cont,
+                                     out));
+    print_result(PARSER_DEV_NUM, TEST_RESULT_SUCCESS);
+    return 0;
+}
+
+int cmd_spi_transfer_bytes(int argc, char **argv)
+{
+    const char* USAGE = "spi_release DEV CS_PORT CS_PIN CONT IN_LEN [OUT0...OUTn]";
```

Comment:
oh cmon

Reply:
lol. Yup I should maybe take a closer look... The args are all ok :smile: